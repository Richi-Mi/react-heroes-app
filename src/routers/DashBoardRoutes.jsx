import React from 'react';
import { Route, Routes } from 'react-router-dom';

import Navbar from '../components/UI/Navbar';
import MarvelScreen from '../components/Marvel/MarvelScreen';
import SearchScreen from '../components/search/SearchScreen';
import DCScreen from '../components/DC/DCScreen';
import Error404 from '../components/UI/Error404';
import Hero from '../components/hero/HeroScreen';
import HomeScreen from '../components/UI/HomeScreen';

const DashBoardRoutes = () => {
    return < >
        <Navbar />
        <div className="container">
            <Routes>
                <Route path='marvel' element={ <MarvelScreen /> } />
                <Route path='dc' element={ <DCScreen />} />
                <Route path='search' element={ <SearchScreen /> } />
                <Route path='hero/:heroeID' element={ <Hero /> } />

                <Route path='/' element={ <HomeScreen />} />
                <Route path='*' element={ <Error404 />} />
            </Routes>
        </div>
    </ >;
};

export default DashBoardRoutes;
