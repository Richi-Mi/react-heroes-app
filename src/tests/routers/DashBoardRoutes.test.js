import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";

import AuthContext from "../../auth/authContext";
import DashBoardRoutes from "../../routers/DashBoardRoutes";

describe(' * Pruebas en el componente <DashBoardRoutes /> ', () => {

    const contextValue = {
        user: {
            logged: true,
            name: 'Jose'
        }
    }
    // Cuando trabajemos en las pruebas y usemos contexto dentro de los componentes
    // tenemos que encerrarlo en un componente Context, y si oucpamos hacer pruebas 
    // con un componente que usa useNavigate() debemos encerrar el componente en 
    // el MemoryRouter de react-router-dom.

    // Usamos mount por que sirve mejor para renderizar todo el componente 
    // no como el shallow

    // propiedad inititialEntries .- Le mandamos un arreglo con el URL al que navegara y renderizara el componente
    

    test(' - Debe de Mostrarse correctamente <HomeScreen />', () => {

        const wrapper = mount( 
            <AuthContext.Provider value={ contextValue }> 
                <MemoryRouter initialEntries={ ['/'] }>
                    <DashBoardRoutes />
                </MemoryRouter>
            </AuthContext.Provider>
        );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('.text_info').text().trim() ).toBe( 'Jose' );
    });

    test(' - Debe de Mostrarse correctamente <MarvelScreen />', () => {

        const wrapper = mount( 
            <AuthContext.Provider value={ contextValue }> 
                <MemoryRouter initialEntries={ ['/marvel'] }>
                    <DashBoardRoutes />
                </MemoryRouter>
            </AuthContext.Provider>
        );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('.text_info').text().trim() ).toBe( 'Jose' );
        expect( wrapper.find('h1').text().trim() ).toBe( 'Marvel Heroes' );
    });
    
    test(' - Debe de Mostrarse correctamente <DCScreen />', () => {
        
        const wrapper = mount( 
            <AuthContext.Provider value={ contextValue }> 
                <MemoryRouter initialEntries={ ['/dc'] }>
                    <DashBoardRoutes />
                </MemoryRouter>
            </AuthContext.Provider>
        );
        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('.text_info').text().trim() ).toBe( 'Jose' );
        expect( wrapper.find('h1').text().trim() ).toBe( 'DC Heroes' );
    });
    
});
