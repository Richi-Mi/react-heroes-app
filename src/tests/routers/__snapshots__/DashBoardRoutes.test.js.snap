// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` * Pruebas en el componente <DashBoardRoutes />   - Debe de Mostrarse correctamente <DCScreen /> 1`] = `
Array [
  <nav
    className="nav"
  >
    <div
      className="nav_options"
    >
      <a
        className="nav_link"
        href="/"
        onClick={[Function]}
      >
        <i
          className="fas fa-home"
        />
        Inicio
      </a>
      <a
        className="nav_link"
        href="/marvel"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        aria-current="page"
        className="nav_link active"
        href="/dc"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        className="nav_link"
        href="/search"
        onClick={[Function]}
      >
        Search
        <i
          className="fas fa-search"
        />
      </a>
    </div>
    <div
      className="nav_acount"
    >
      <span
        className="text_info acount_item"
      >
        Jose
      </span>
      <button
        className="btn_logout acount_item"
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
  </nav>,
  <div
    className="container"
  >
    Array [
      <h1>
         DC Heroes 
      </h1>,
      <hr />,
      <div
        className="card_heroes"
      >
        <div
          className="card animate__animated animate__fadeIn"
        >
          <div
            className="card_img"
          >
            <img
              alt="Batman"
              src="/assets/dc-batman.jpg"
            />
          </div>
          <h3
            className="dc"
          >
             
            Batman
             
          </h3>
          <div
            className="card_info"
          >
            <div
              className="card_text"
            >
              <p>
                 
                Bruce Wayne
                 
              </p>
            </div>
            <div
              className="card_link"
            >
              <a
                className="link"
                href="/hero/dc-batman"
                onClick={[Function]}
              >
                Mas 
                <i
                  className="fas fa-arrow-right"
                />
              </a>
            </div>
          </div>
        </div>
        <div
          className="card animate__animated animate__fadeIn"
        >
          <div
            className="card_img"
          >
            <img
              alt="Superman"
              src="/assets/dc-superman.jpg"
            />
          </div>
          <h3
            className="dc"
          >
             
            Superman
             
          </h3>
          <div
            className="card_info"
          >
            <div
              className="card_text"
            >
              <p>
                 
                Kal-El
                 
              </p>
            </div>
            <div
              className="card_link"
            >
              <a
                className="link"
                href="/hero/dc-superman"
                onClick={[Function]}
              >
                Mas 
                <i
                  className="fas fa-arrow-right"
                />
              </a>
            </div>
          </div>
        </div>
        <div
          className="card animate__animated animate__fadeIn"
        >
          <div
            className="card_img"
          >
            <img
              alt="Flash"
              src="/assets/dc-flash.jpg"
            />
          </div>
          <h3
            className="dc"
          >
             
            Flash
             
          </h3>
          <div
            className="card_info"
          >
            <div
              className="card_text"
            >
              <p>
                 
                Jay Garrick
                 
              </p>
            </div>
            <div
              className="card_link"
            >
              <a
                className="link"
                href="/hero/dc-flash"
                onClick={[Function]}
              >
                Mas 
                <i
                  className="fas fa-arrow-right"
                />
              </a>
            </div>
          </div>
        </div>
        <div
          className="card animate__animated animate__fadeIn"
        >
          <div
            className="card_img"
          >
            <img
              alt="Green Lantern"
              src="/assets/dc-green.jpg"
            />
          </div>
          <h3
            className="dc"
          >
             
            Green Lantern
             
          </h3>
          <div
            className="card_info"
          >
            <div
              className="card_text"
            >
              <p>
                 
                Alan Scott
                 
              </p>
            </div>
            <div
              className="card_link"
            >
              <a
                className="link"
                href="/hero/dc-green"
                onClick={[Function]}
              >
                Mas 
                <i
                  className="fas fa-arrow-right"
                />
              </a>
            </div>
          </div>
        </div>
        <div
          className="card animate__animated animate__fadeIn"
        >
          <div
            className="card_img"
          >
            <img
              alt="Green Arrow"
              src="/assets/dc-arrow.jpg"
            />
          </div>
          <h3
            className="dc"
          >
             
            Green Arrow
             
          </h3>
          <div
            className="card_info"
          >
            <div
              className="card_text"
            >
              <p>
                 
                Oliver Queen
                 
              </p>
            </div>
            <div
              className="card_link"
            >
              <a
                className="link"
                href="/hero/dc-arrow"
                onClick={[Function]}
              >
                Mas 
                <i
                  className="fas fa-arrow-right"
                />
              </a>
            </div>
          </div>
        </div>
        <div
          className="card animate__animated animate__fadeIn"
        >
          <div
            className="card_img"
          >
            <img
              alt="Wonder Woman"
              src="/assets/dc-wonder.jpg"
            />
          </div>
          <h3
            className="dc"
          >
             
            Wonder Woman
             
          </h3>
          <div
            className="card_info"
          >
            <div
              className="card_text"
            >
              <p>
                 
                Princess Diana
                 
              </p>
            </div>
            <div
              className="card_link"
            >
              <a
                className="link"
                href="/hero/dc-wonder"
                onClick={[Function]}
              >
                Mas 
                <i
                  className="fas fa-arrow-right"
                />
              </a>
            </div>
          </div>
        </div>
        <div
          className="card animate__animated animate__fadeIn"
        >
          <div
            className="card_img"
          >
            <img
              alt="Martian Manhunter"
              src="/assets/dc-martian.jpg"
            />
          </div>
          <h3
            className="dc"
          >
             
            Martian Manhunter
             
          </h3>
          <div
            className="card_info"
          >
            <div
              className="card_text"
            >
              <p>
                 
                J'onn J'onzz
                 
              </p>
            </div>
            <div
              className="card_link"
            >
              <a
                className="link"
                href="/hero/dc-martian"
                onClick={[Function]}
              >
                Mas 
                <i
                  className="fas fa-arrow-right"
                />
              </a>
            </div>
          </div>
        </div>
        <div
          className="card animate__animated animate__fadeIn"
        >
          <div
            className="card_img"
          >
            <img
              alt="Robin/Nightwing"
              src="/assets/dc-robin.jpg"
            />
          </div>
          <h3
            className="dc"
          >
             
            Robin/Nightwing
             
          </h3>
          <div
            className="card_info"
          >
            <div
              className="card_text"
            >
              <p>
                 
                Dick Grayson
                 
              </p>
            </div>
            <div
              className="card_link"
            >
              <a
                className="link"
                href="/hero/dc-robin"
                onClick={[Function]}
              >
                Mas 
                <i
                  className="fas fa-arrow-right"
                />
              </a>
            </div>
          </div>
        </div>
        <div
          className="card animate__animated animate__fadeIn"
        >
          <div
            className="card_img"
          >
            <img
              alt="Blue Beetle"
              src="/assets/dc-blue.jpg"
            />
          </div>
          <h3
            className="dc"
          >
             
            Blue Beetle
             
          </h3>
          <div
            className="card_info"
          >
            <div
              className="card_text"
            >
              <p>
                 
                Dan Garret
                 
              </p>
            </div>
            <div
              className="card_link"
            >
              <a
                className="link"
                href="/hero/dc-blue"
                onClick={[Function]}
              >
                Mas 
                <i
                  className="fas fa-arrow-right"
                />
              </a>
            </div>
          </div>
        </div>
        <div
          className="card animate__animated animate__fadeIn"
        >
          <div
            className="card_img"
          >
            <img
              alt="Black Canary"
              src="/assets/dc-black.jpg"
            />
          </div>
          <h3
            className="dc"
          >
             
            Black Canary
             
          </h3>
          <div
            className="card_info"
          >
            <div
              className="card_text"
            >
              <p>
                 
                Dinah Drake
                 
              </p>
            </div>
            <div
              className="card_link"
            >
              <a
                className="link"
                href="/hero/dc-black"
                onClick={[Function]}
              >
                Mas 
                <i
                  className="fas fa-arrow-right"
                />
              </a>
            </div>
          </div>
        </div>
      </div>,
    ]
  </div>,
]
`;

exports[` * Pruebas en el componente <DashBoardRoutes />   - Debe de Mostrarse correctamente <HomeScreen /> 1`] = `
Array [
  <nav
    className="nav"
  >
    <div
      className="nav_options"
    >
      <a
        aria-current="page"
        className="nav_link active"
        href="/"
        onClick={[Function]}
      >
        <i
          className="fas fa-home"
        />
        Inicio
      </a>
      <a
        className="nav_link"
        href="/marvel"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        className="nav_link"
        href="/dc"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        className="nav_link"
        href="/search"
        onClick={[Function]}
      >
        Search
        <i
          className="fas fa-search"
        />
      </a>
    </div>
    <div
      className="nav_acount"
    >
      <span
        className="text_info acount_item"
      >
        Jose
      </span>
      <button
        className="btn_logout acount_item"
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div
      className="content_home animate__animated animate__fadeIn"
    >
      <h2
        className="title_home"
      >
         Heroes App 
      </h2>
      <h3
        className="subtitle_home"
      >
         A list Heroes of Comics 
      </h3>
      <div
        className="gif_home"
      >
        <img
          alt="heroes"
          src="/banner_code.jpg"
        />
      </div>
      <div
        className="btns"
      >
        <a
          className="link_more"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel Heroes
        </a>
        <a
          className="link_more"
          href="/dc"
          onClick={[Function]}
        >
          DC Heroes
        </a>
      </div>
    </div>
  </div>,
]
`;

exports[` * Pruebas en el componente <DashBoardRoutes />   - Debe de Mostrarse correctamente <MarvelScreen /> 1`] = `
Array [
  <nav
    className="nav"
  >
    <div
      className="nav_options"
    >
      <a
        className="nav_link"
        href="/"
        onClick={[Function]}
      >
        <i
          className="fas fa-home"
        />
        Inicio
      </a>
      <a
        aria-current="page"
        className="nav_link active"
        href="/marvel"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        className="nav_link"
        href="/dc"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        className="nav_link"
        href="/search"
        onClick={[Function]}
      >
        Search
        <i
          className="fas fa-search"
        />
      </a>
    </div>
    <div
      className="nav_acount"
    >
      <span
        className="text_info acount_item"
      >
        Jose
      </span>
      <button
        className="btn_logout acount_item"
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
  </nav>,
  <div
    className="container"
  >
    Array [
      <h1>
         Marvel Heroes 
      </h1>,
      <hr />,
      <div
        className="card_heroes"
      >
        <div
          className="card animate__animated animate__fadeIn"
        >
          <div
            className="card_img"
          >
            <img
              alt="Spider Man"
              src="/assets/marvel-spider.jpg"
            />
          </div>
          <h3
            className="marvel"
          >
             
            Spider Man
             
          </h3>
          <div
            className="card_info"
          >
            <div
              className="card_text"
            >
              <p>
                 
                Peter Parker
                 
              </p>
            </div>
            <div
              className="card_link"
            >
              <a
                className="link"
                href="/hero/marvel-spider"
                onClick={[Function]}
              >
                Mas 
                <i
                  className="fas fa-arrow-right"
                />
              </a>
            </div>
          </div>
        </div>
        <div
          className="card animate__animated animate__fadeIn"
        >
          <div
            className="card_img"
          >
            <img
              alt="Captain America"
              src="/assets/marvel-captain.jpg"
            />
          </div>
          <h3
            className="marvel"
          >
             
            Captain America
             
          </h3>
          <div
            className="card_info"
          >
            <div
              className="card_text"
            >
              <p>
                 
                Steve Rogers
                 
              </p>
            </div>
            <div
              className="card_link"
            >
              <a
                className="link"
                href="/hero/marvel-captain"
                onClick={[Function]}
              >
                Mas 
                <i
                  className="fas fa-arrow-right"
                />
              </a>
            </div>
          </div>
        </div>
        <div
          className="card animate__animated animate__fadeIn"
        >
          <div
            className="card_img"
          >
            <img
              alt="Iron Man"
              src="/assets/marvel-iron.jpg"
            />
          </div>
          <h3
            className="marvel"
          >
             
            Iron Man
             
          </h3>
          <div
            className="card_info"
          >
            <div
              className="card_text"
            >
              <p>
                 
                Tony Stark
                 
              </p>
            </div>
            <div
              className="card_link"
            >
              <a
                className="link"
                href="/hero/marvel-iron"
                onClick={[Function]}
              >
                Mas 
                <i
                  className="fas fa-arrow-right"
                />
              </a>
            </div>
          </div>
        </div>
        <div
          className="card animate__animated animate__fadeIn"
        >
          <div
            className="card_img"
          >
            <img
              alt="Thor"
              src="/assets/marvel-thor.jpg"
            />
          </div>
          <h3
            className="marvel"
          >
             
            Thor
             
          </h3>
          <div
            className="card_info"
          >
            <div
              className="card_text"
            >
              <p>
                 
                Thor Odinson
                 
              </p>
            </div>
            <div
              className="card_link"
            >
              <a
                className="link"
                href="/hero/marvel-thor"
                onClick={[Function]}
              >
                Mas 
                <i
                  className="fas fa-arrow-right"
                />
              </a>
            </div>
          </div>
        </div>
        <div
          className="card animate__animated animate__fadeIn"
        >
          <div
            className="card_img"
          >
            <img
              alt="Hulk"
              src="/assets/marvel-hulk.jpg"
            />
          </div>
          <h3
            className="marvel"
          >
             
            Hulk
             
          </h3>
          <div
            className="card_info"
          >
            <div
              className="card_text"
            >
              <p>
                 
                Bruce Banner
                 
              </p>
            </div>
            <div
              className="card_link"
            >
              <a
                className="link"
                href="/hero/marvel-hulk"
                onClick={[Function]}
              >
                Mas 
                <i
                  className="fas fa-arrow-right"
                />
              </a>
            </div>
          </div>
        </div>
        <div
          className="card animate__animated animate__fadeIn"
        >
          <div
            className="card_img"
          >
            <img
              alt="Wolverine"
              src="/assets/marvel-wolverine.jpg"
            />
          </div>
          <h3
            className="marvel"
          >
             
            Wolverine
             
          </h3>
          <div
            className="card_info"
          >
            <div
              className="card_text"
            >
              <p>
                 
                James Howlett
                 
              </p>
            </div>
            <div
              className="card_link"
            >
              <a
                className="link"
                href="/hero/marvel-wolverine"
                onClick={[Function]}
              >
                Mas 
                <i
                  className="fas fa-arrow-right"
                />
              </a>
            </div>
          </div>
        </div>
        <div
          className="card animate__animated animate__fadeIn"
        >
          <div
            className="card_img"
          >
            <img
              alt="Daredevil"
              src="/assets/marvel-daredevil.jpg"
            />
          </div>
          <h3
            className="marvel"
          >
             
            Daredevil
             
          </h3>
          <div
            className="card_info"
          >
            <div
              className="card_text"
            >
              <p>
                 
                Matthew Murdock
                 
              </p>
            </div>
            <div
              className="card_link"
            >
              <a
                className="link"
                href="/hero/marvel-daredevil"
                onClick={[Function]}
              >
                Mas 
                <i
                  className="fas fa-arrow-right"
                />
              </a>
            </div>
          </div>
        </div>
        <div
          className="card animate__animated animate__fadeIn"
        >
          <div
            className="card_img"
          >
            <img
              alt="Hawkeye"
              src="/assets/marvel-hawkeye.jpg"
            />
          </div>
          <h3
            className="marvel"
          >
             
            Hawkeye
             
          </h3>
          <div
            className="card_info"
          >
            <div
              className="card_text"
            >
              <p>
                 
                Clinton Barton
                 
              </p>
            </div>
            <div
              className="card_link"
            >
              <a
                className="link"
                href="/hero/marvel-hawkeye"
                onClick={[Function]}
              >
                Mas 
                <i
                  className="fas fa-arrow-right"
                />
              </a>
            </div>
          </div>
        </div>
        <div
          className="card animate__animated animate__fadeIn"
        >
          <div
            className="card_img"
          >
            <img
              alt="Cyclops"
              src="/assets/marvel-cyclops.jpg"
            />
          </div>
          <h3
            className="marvel"
          >
             
            Cyclops
             
          </h3>
          <div
            className="card_info"
          >
            <div
              className="card_text"
            >
              <p>
                 
                Scott Summers
                 
              </p>
            </div>
            <div
              className="card_link"
            >
              <a
                className="link"
                href="/hero/marvel-cyclops"
                onClick={[Function]}
              >
                Mas 
                <i
                  className="fas fa-arrow-right"
                />
              </a>
            </div>
          </div>
        </div>
        <div
          className="card animate__animated animate__fadeIn"
        >
          <div
            className="card_img"
          >
            <img
              alt="Silver Surfer"
              src="/assets/marvel-silver.jpg"
            />
          </div>
          <h3
            className="marvel"
          >
             
            Silver Surfer
             
          </h3>
          <div
            className="card_info"
          >
            <div
              className="card_text"
            >
              <p>
                 
                Norrin Radd
                 
              </p>
            </div>
            <div
              className="card_link"
            >
              <a
                className="link"
                href="/hero/marvel-silver"
                onClick={[Function]}
              >
                Mas 
                <i
                  className="fas fa-arrow-right"
                />
              </a>
            </div>
          </div>
        </div>
      </div>,
    ]
  </div>,
]
`;
